.github/workflows/release.yml

```diff
@@ -0,0 +1,45 @@
+name: release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  build:
+    runs-on: macos-latest
+
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v2
+
+      - name: Set up Swift
+        uses: swift-actions/setup-swift@v2
+        with:
+          swift-version: '6.0'
+
+      - name: Build a universal binary
+        run: swift build --configuration release --arch arm64 --arch x86_64
+
+      # TODO: Replace with maintained actions or GitHub CLI calls
+
+      - name: Create a release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ github.ref }}
+          draft: false
+          prerelease: false
+
+      - name: Upload the built binary
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: .build/release/ql
+          asset_name: ql
+          asset_content_type: application/octet-stream
```

Comment:
This can probably be replaced with:

1. [`gh release create [<tag>] [<files>...]`](https://cli.github.com/manual/gh_release_create)
2. [`gh release upload <tag> <files>... [flags]`](https://cli.github.com/manual/gh_release_upload)